build-system/status-page/incident.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview
+ * Sync status.amp.dev with cherry-pick progress
+ */
+
+const [number, body, before, after] = process.argv.slice(2);
+const fetch = require('node-fetch');
+const {getChannels, getFormats, steps} = require('./common');
+const {log} = require('../common/logging');
+const apiUrl = `https://api.statuspage.io/v1/pages/${process.env.STATUS_PAGE_ID}`;
+const headers = {
+  'Content-Type': 'application/json',
+  'Authorization': `OAuth ${process.env.STATUS_PAGE_TOKEN}`,
+};
+const componentsIds = {
+  'Websites': process.env.WEBSITES_ID || '7sg50qz95rt7',
+  'Stories': process.env.STORIES_ID || 'yy777zy1tzt4',
+  'Ads': process.env.ADS_ID || '9vglh2hndltg',
+  'Emails': process.env.EMAILS_ID || 'd3lmsf61qb52',
+};
+const updateBodies = {
+  'identified': 'The issue has been identified and a fix is underway.',
+  'monitoring': `The fix has been deployed and is being rolled out to the CDN.
+    Please allow up to 30 minutes for the CDN to pick up the fix.`,
+  'resolved': 'The fix has been verified.',
+};
+
+/**
+ * Create incident on status.amp.dev
+ * @param {Array<string>} channels
+ * @param {Array<string>} formats
+ * @param {string} status
+ * @return {Promise<Object>} response
```

Comment:
Same suggestion here about adding a typedef for the object type.

Also a minor nit: `comment.js` is using lowercase `object` in the type while this file isn't.

Reply:
re: typedef, do you mean writing an interface for statuspage.io API's json responses? For example, this is a sample response when creating an incident.. I don't think it's worth the effort when all we do is log it.
```
{
"id": "xby4tnnxnsn9",
"components": [
{}
],
"created_at": "2021-07-07T18:03:05Z",
"impact": "critical",
"impact_override": "minor",
"incident_updates": [
{}
],
"metadata": {
"jira": {}
},
"monitoring_at": "2021-07-07T18:03:05Z",
"name": "Data Layer Migration",
"page_id": "string",
"postmortem_body": "##### Issue\nAt approximately 17:02 UTC on 2013-04-21, our master database server unexpectedly went unresponsive to all network.\nA reboot of the machine at 17:05 UTC resulted in a failed mount of a corrupted EBS volume, and we made the decision\nat that time to fail over the slave database.\n\n##### Resolution\nAt 17:12 UTC, the slave database had been successfully promoted to master and the application recovered enough to\naccept web traffic again. A new slave database node was created and placed into the rotation to guard against future\nmaster failures. The promoted slave database performed slowly for the next couple of hours as the query cache began\nto warm up, and eventually settled into a reasonable performance profile around 20:00 UTC.\n\n##### Future Mitigation Plans\nOver the past few months, we have been working on an overhaul to our data storage layer with a migration from a Postgres\nsetup to a distributed, fault-tolerant, multi-region data layer using Riak. This initiative has been prioritized, and\nthe migration will be performed in the coming weeks. We will notify our clients of the scheduled downtime via an\nincident on this status site, and via a blog post.\n",
"postmortem_body_last_updated_at": "2021-07-07T18:03:05Z",
"postmortem_ignored": true,
"postmortem_notified_subscribers": true,
"postmortem_notified_twitter": true,
"postmortem_published_at": true,
"resolved_at": "2021-07-07T18:03:05Z",
"scheduled_auto_completed": true,
"scheduled_auto_in_progress": true,
"scheduled_for": "2013-05-07T03:00:00.007Z",
"scheduled_remind_prior": true,
"scheduled_reminded_at": "2021-07-07T18:03:05Z",
"scheduled_until": "2013-05-07T06:00:00.007Z",
"shortlink": "http://stspg.io/803310a12",
"status": "scheduled",
"updated_at": "2021-07-07T18:03:05Z"
}
```

Reply:
Yep, that's what I meant. And I agree it's not worth documenting every field if the response is merely being logged.